Gecode/gecode
# release/6.3.0

- Bumped to 6.2.1
- Configure whether to use propagation and/or failure for action
- Removed shadowing
- Fixed linear cutoff generator
- Small fixes and split out job shop instances (including random ones)
- Small fixes and split out job shop instances (including random ones)
- Extended types for statistics to avoid wraparound for long-running benchmarks
- Extended types for statistics to avoid wraparound for long-running benchmarks
- Compile fix
- Update FlatZinc parser error message for out of bound integer limits (#48)
- Fix bool-set channel constraint when set variable is already fixed during posting
- Only use symmetry breaking if not using CDBF
- Compiles with MSVC 2015 again
- Add Florian's details
- Add missing include of sys/socket.h
- Use clang 8 for travis
- Use clang 7 for travis
- Reverted to clang 3.8 for travis
- Upgrade Travis (#54)
- Ignore Cmake files
- Use unordered_map even if build with CMake
- Turn macros into expressions
- Do not use NULL
- Updated
- Use noexcept instead of throw()
- Delete member functions rather than making them private
- Removed some warning, fixed some minor quirks
- Solutions where not properly counted as unsigned long long...
- Warnings
- Fixed another long long confusion for restart cutoffs
- Fixed another long long confusion for restart cutoffs
- Doc typos
- More long long statistics glitches fixed
- Resolved overloading
- Warnings removed
- Better
- Use a separate thread for acquiring and releasing mutexes that have t… (#55)
- Added test for propagator groups
- Fixed crash after PropagatorGroup::kill
- Improved propagator group tests
- Use min and max instead of val for FloatMinimize/MaximizeSpace
- Update MiniZinc globals library to new (MiniZinc 2.3) structure, to take advantage of decompositions for reified constraints for which there is no special propagator in Gecode.
- Fix cumulative for zero duration tasks
- Fix int_set_channel and roots for empty sets
- Change log
- Fix registry for counting constraints
- Rewrite disjunctive to fzn_cumulative rather than cumulative to avoid infinite loop
- Ensure var name arrays match after shrinking
- Set the correct policy for building with Gist in CMake3.17+
- Add .editorconfig
- Better FlatZinc error messages
- Avoid double underscorde in include guards
- Extend changelog format for GitHub issues
- Add changelog-entry for double underscore guards
- Changelog entry for better flatzinc errors
- Add missing FlatZinc section to text changelog
- Use UTF-8 as file encoding
- Make make check work when no dynamic load path exists
- Minor, autogenerated changes and line endings
- Make elements and weights in OverweightValues non-const
- Add sum_set to gecode_set_weights conversion. Fixes #49.
- Fix fzn_table_bool and fzn_table_bool_reif predicates, which had incorrect types. Fixes #61.
- Fix memory manager to only reuse memory that is properly aligned
- Add configure flags --with-freelist32-size-max and --with-freelist64-size-max to avoid hard-coding these parameters
- Avoid reserved identifier names
- Avoid double underscore in macros
- Suppress erroneous clang warning for array access
- Minor, remove warning from flatzinc.hh
- No reification test for empty constraint
- Change NULL to nullptr
- Minor, clean-ups in test
- Use free instead of delete
- Cast to storage type before multiplication
- Fix path compression in disjoint sets for sorted
- Improve comments in sortsup.hpp
- Remove non-needed cast
- Empty block body comments
- Updated version to 6.3.0, soversion to 51
- Fix FlatZinc parser generation for Bison 3.7
- Update makedepend.perl to search in current directory for quoted-includes
- Avoid unknown pragma warnings for gcc and clang (#88)
- Fix testing for IntAction, BoolAction, SetAction, FloatAction with merit functions
- Avoid unknown pragma warnings for msvc
- Add disabled pragma warnings to cmake
- Use single option to activate cp-profiler (#90)
- Fix compilation errors for CPProfiler support
- Add warning not to use copy constructor and copy directly
- Remove empty statement warnings
- Separate test filtering from running
- Refactor Assignment operator++ to next()
- Refactor Assignments operator() to has_more
- Refactoring: Thread rand through calls in test
- Refactor, extract run_tests function
- Refactor, move data and logic into Options
- Refactor, extracted run_test function
- Add support for parallel testing
- Use atomic for propagator identifier
- Make Region pool thread local
- Minor, remove exraneous newline in changelog
- Require C++11
- Dynamically adjust test batch sizes
- Use std::vector instead of std::array
- Add offset versions of element constraints and enable them for FlatZinc
- Minor, typo
- Fix construction of error message when there is no annotation
- Fix FlatZinc 2d element constraint construction
- Fix 2d bool element constraint
- Ready fzn-gecode as MiniZinc solver on `make install`
- Fix c++20 compatibility issue in fetch_add invocation in gpi.hh
- Remove unused variables in float test code.
- Update LICENSE
- Update LICENSE
- Fix MOC invocation in gist folder by bumping cmake_minimum_required to 3.17 because CMP0100 policy prevents .hh files processing before 3.17
- Fix include directory order for CMake compilation
- Fix FlatZinc posting for element constraints
- Update Gist standalone TestSpace to modern Gecode
- uses CMAKE_INSTALL_LIBDIR when installing lib files
- Correctly include GNUInstallDirs and use where relevant
- Fix mistake in directory of MiniZinc library
- Set minimum required CMake version to 3.7.0
- Fix action branching for flatzinc
- Avoid reading uninitialized varaible in distinct
- Fix help message for test binary
- Add RestartStop
- Added default copy assignment operators where their implicit definition is now (C++11) deprecated due to the presence of user-declared copy constructors.
- Added default copy constructors where their implicit definition is now (C++11) deprecated due to the presence of user-declared copy assignment operators. This prevents -Wdeprecated-copy from complaining.
- Add MiniZinc solver configuration for Gecode Gist
- Add support for value_propagation, domain_propagation and bounds_propagation annotations
- Add support for disjunctive and cumulative on optional variables
- Add support for Qt6
- Replace check for C++11 with check for C++17, and accept Qt versions >= 5 (instead of =5)
- Replace using statement with typedef to fix Windows compilation
- Fix compilation without threads
